src/utils.jl

```diff
@@ -494,3 +494,30 @@ end
 @inline __diag(x::AbstractMatrix) = diag(x)
 @inline __diag(x::AbstractVector) = x
 @inline __diag(x::Number) = x
+
+#functions for updating alpha for PseudoTransient
+function switched_evolution_relaxation(alpha::Number,
+        res_norm::Number,
+        nsteps::Int,
+        u,
+        u_prev,
+        fu,
+        norm::F) where {F}
+    new_norm = norm(fu)
+    return alpha * (res_norm / new_norm)
+end
+
+function robust_update_alpha(alpha::Number,
+        res_norm::Number,
+        nsteps::Int,
+        u,
+        u_prev,
+        fu,
+        norm::F) where {F}
+    if nsteps ≤ 100
+        return alpha
```

Comment:
Why `100`? Is this supported by benchmarks? I would expose this as an option via RobustPT.

Reply:
Yes. I played around with threshold and 100 is the best option. I think the user can define their own function easily if they want something else. I don't see how I can expose the threshold without affecting signficant portion of the pt methods.

Reply:
Not really, it should affect PT at all. Create a closure with the outer function taking `n` as input, or use partial functions